agents/supervisor/supervisor.go

```diff
@@ -52,9 +53,12 @@ type Supervisor struct {
 	qanRequests   chan agentpb.QANCollectRequest
 	l             *logrus.Entry
 
-	m              sync.Mutex
+	rw             sync.RWMutex
 	agentProcesses map[string]*agentProcessInfo
 	builtinAgents  map[string]*builtinAgentInfo
+
+	amx           sync.Mutex
```

Comment:
It looks like if we use RWMutex, we can have parallel `/local/Status` calls which is a good thing.